### ISSUE TITLE
Dimensionality Reduction Technique

### ISSUE THREAD

**Issue by davideaurucci** [RUG-IDS/Assignments#25]
Hi,

In the last question of the first exercise of the assignment 6 the pdf says: "Use a suitable dimensionality reduction technique to make a scatterplot of the texts based on word-occurrences. Can you visually observe clusters of similar texts? Can you spot outliers?"

How do you mean that? Should we use any feature extraction/selection technique?
We were thinking to use LDA in order to find the most relevant topics in the corpus (state_union corpus), but we would not know how to plot the results in a scatter-plot in order to find cluster similarities/differences.

Can we have an hint about that?

Thanks for your attention

team 13

**Comment by HanKruiger** [comment]
Good question, and I agree that the exercise is a bit open-ended.

Extracting the (two) most relevant topics in the entire corpus *is* an option, but arguably not the best one. If you were to do this, you would have the two values for every speech: x and y. Where x is the frequency of relevant topic 1, and y is the frequency of relevant topic 2. You can then simply scatter all speeches by their x- and y-values. The advantage of this option is that the x- and y-axes have a very clear meaning, namely the two word-frequencies. A disadvantage is that some (or many) speeches may not even mention the two topics, and will be placed at the origin of the scatterplot, so finding clusters/outliers is harder.

A better option would be to compute [cosine-distances](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.cosine_distances.html#sklearn.metrics.pairwise.cosine_distances) for all pairs of speeches s.t. you have a square matrix with elements d_ij, where d_ij is the cosine distance between speech i and speech j. This can be used as input for a multidimensional scaling ([MDS](https://en.wikipedia.org/wiki/Multidimensional_scaling)) implementation (e.g. [this one](http://scikit-learn.org/stable/modules/generated/sklearn.manifold.MDS.html#sklearn.manifold.MDS)), which outputs a set of 2D coordinates that can be easily scattered.
A disadvantage of this option is that the x- and y-axes don't have a clear interpretation. However you *should* be able to spot clusters and outliers more easily.
